SQL_Queries.txt

```diff
@@ -0,0 +1,1670 @@
+| 1001 | Peel    | 3008 | 10/05/90 |    4723.00 |
+| 1003 | AxelRod | 3009 | 10/04/90 |    1713.23 |
+| 1002 | Serres  | 3010 | 10/06/90 |    1309.95 |
+| 1001 | Peel    | 3011 | 10/06/90 |    9891.88 |
++------+---------+------+----------+------------+
+
+
+4) Arrange the Order table by descending customer number.
+
+SELECT * FROM orders order by cnum desc ;
+output:
++------+---------+----------+------+------+
+| ONUM | AMT     | ODATE    | CNUM | SNUM |
++------+---------+----------+------+------+
+| 3001 |   18.69 | 10/03/90 | 2008 | 1007 |
+| 3006 | 1098.16 | 10/03/90 | 2008 | 1007 |
+| 3002 | 1900.10 | 10/03/90 | 2007 | 1004 |
+| 3008 | 4723.00 | 10/05/90 | 2006 | 1001 |
+| 3011 | 9891.88 | 10/06/90 | 2006 | 1001 |
+| 3007 |   75.75 | 10/04/90 | 2004 | 1002 |
+| 3010 | 1309.95 | 10/06/90 | 2004 | 1002 |
+| 3005 | 5160.45 | 10/03/90 | 2003 | 1002 |
+| 3009 | 1713.23 | 10/04/90 | 2002 | 1003 |
+| 3003 |  767.19 | 10/03/90 | 2001 | 1001 |
++------+---------+----------+------+------+
+10 rows in set (0.00 sec)
+
+5) Find which salespeople currently have orders in the order table.
+
+// Without Join
+Select * from salespeople where snum in (select snum from orders) ;
+output:
++------+---------+-----------+------+
+| SNUM | SNAME   | CITY      | COMM |
++------+---------+-----------+------+
+| 1001 | Peel    | London    |   12 |
+| 1002 | Serres  | Sanjose   |   13 |
+| 1003 | AxelRod | New York  |   10 |
+| 1004 | Motika  | London    |   11 |
+| 1007 | RifKin  | Barcelona |   15 |
++------+---------+-----------+------+
+5 rows in set (0.00 sec)
+
+// With Join
+select s.* from salespeople as s
+inner join
+orders as o where o.snum = s.snum group by s.snum ;
+output:
++------+---------+-----------+------+
+| SNUM | SNAME   | CITY      | COMM |
++------+---------+-----------+------+
+| 1001 | Peel    | London    |   12 |
+| 1002 | Serres  | Sanjose   |   13 |
+| 1003 | AxelRod | New York  |   10 |
+| 1004 | Motika  | London    |   11 |
+| 1007 | RifKin  | Barcelona |   15 |
++------+---------+-----------+------+
+5 rows in set (0.00 sec)
+
+
+6) List names of all customers matched with the salespeople serving them.
+
+ SELECT c.cname AS customer , s.sname AS ServingBy from customers c
+ INNER JOIN
+ Salespeople s where c.snum = s.snum;
+ output:
++----------+-----------+
+| customer | ServingBy |
++----------+-----------+
+| Hoffman  | Peel      |
+| Giovanni | AxelRod   |
+| Liu      | Serres    |
+| Grass    | Serres    |
+| Clemens  | Peel      |
+| Pereira  | Motika    |
+| Cisneros | RifKin    |
++----------+-----------+
+
+7) Find the names and numbers of all salespeople who have more than one customer.
+
+select sname , snum from salespeople where snum in (select snum from customers group by snum having count(snum) > 1 );
+output:
++--------+------+
+| sname  | snum |
++--------+------+
+| Peel   | 1001 |
+| Serres | 1002 |
++--------+------+
+
+8) Count the orders of each of the salespeople and output the results in descending order. 
```

Comment:
Query will not work if there is salespeople who is not associated with any order 
Expected
In that case the order count of salespeople  show as 0.